File: python/Komputer.py
from Motherboard import Motherboard
from CPU import CPU
from GPU import GPU
from RAM import RAM
from Penyimpanan import Penyimpanan

class Komputer:
    def __init__(self, nama="", motherboard=Motherboard(), cpu=CPU(), listRAM=[], gpu=GPU(), penyimpanan=Penyimpanan()):
        self.nama = nama
        self.motherboard = motherboard
        self.cpu = cpu
        self.listRAM = listRAM
        self.gpu = gpu
        self.penyimpanan = penyimpanan
        
        self.total_harga = self.motherboard.getHarga() + self.cpu.getHarga() + self.gpu.getHarga() + self.penyimpanan.getHarga()
        for ram in self.listRAM:
            self.total_harga += ram.getHarga()
    
    def setNama(self, nama):
        self.nama = nama
    
    def setMotherboard(self, motherboard):
        self.motherboard = motherboard
        self.updateTotalHarga()
    
    def setCPU(self, cpu):
        self.cpu = cpu
        self.updateTotalHarga()
    
    def setListRAM(self, listRAM):
        self.listRAM = listRAM
        self.updateTotalHarga()
    
    def setGPU(self, gpu):
        self.gpu = gpu
        self.updateTotalHarga()
    
    def setPenyimpanan(self, penyimpanan):
        self.penyimpanan = penyimpanan
        self.updateTotalHarga()
    
    def addRAM(self, ram):
        self.listRAM.append(ram)
        self.total_harga += ram.getHarga()
    
    def getNama(self):
        return self.nama
    
    def getMotherboard(self):
        return self.motherboard
    
    def getCPU(self):
        return self.cpu
    
    def getListRAM(self):
        return self.listRAM
    
    def getGPU(self):
        return self.gpu
    
    def getPenyimpanan(self):
        return self.penyimpanan
    
    def getTotalHarga(self):
        return self.total_harga
    
    def updateTotalHarga(self):
        self.total_harga = self.motherboard.getHarga() + self.cpu.getHarga() + self.gpu.getHarga() + self.penyimpanan.getHarga()
        for ram in self.listRAM:
            self.total_harga += ram.getHarga()

File: cpp/Komputer.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Motherboard.cpp"
#include "CPU.cpp"
#include "GPU.cpp"
#include "RAM.cpp"
#include "Penyimpanan.cpp"
using namespace std;

class Komputer
{
private:
    string nama;
    Motherboard motherboard;
    CPU cpu;
    vector<RAM> listRAM;
    GPU gpu;
    Penyimpanan penyimpanan;
    int total_harga;

public:
    Komputer() {}

    Komputer(string nama, Motherboard motherboard, CPU cpu, vector<RAM> listRAM, GPU gpu, Penyimpanan penyimpanan)
    {
        this->nama = nama;
        this->motherboard = motherboard;
        this->cpu = cpu;
        this->listRAM = listRAM;
        this->gpu = gpu;
        this->penyimpanan = penyimpanan;

        // Menghitung total harga komponen
        this->total_harga = motherboard.getHarga() + cpu.getHarga() + gpu.getHarga() + penyimpanan.getHarga();
        for (RAM ram : listRAM)
        {
            this->total_harga += ram.getHarga();
        }
    }

    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setMotherboard(Motherboard motherboard)
    {
        this->motherboard = motherboard;
        updateTotalHarga();
    }

    void setCPU(CPU cpu)
    {
        this->cpu = cpu;
        updateTotalHarga();
    }

    void setListRAM(vector<RAM> listRAM)
    {
        this->listRAM = listRAM;
        updateTotalHarga();
    }

    void setGPU(GPU gpu)
    {
        this->gpu = gpu;
        updateTotalHarga();
    }

    void setPenyimpanan(Penyimpanan penyimpanan)
    {
        this->penyimpanan = penyimpanan;
        updateTotalHarga();
    }

    void addRAM(RAM ram)
    {
        this->listRAM.push_back(ram);
        this->total_harga += ram.getHarga();
    }

    string getNama()
    {
        return nama;
    }

    Motherboard getMotherboard()
    {
        return motherboard;
    }

    CPU getCPU()
    {
        return cpu;
    }

    vector<RAM> getListRAM()
    {
        return listRAM;
    }

    GPU getGPU()
    {
        return gpu;
    }

    Penyimpanan getPenyimpanan()
    {
        return penyimpanan;
    }

    int getTotalHarga()
    {
        return total_harga;
    }

    void updateTotalHarga()
    {
        this->total_harga = motherboard.getHarga() + cpu.getHarga() + gpu.getHarga() + penyimpanan.getHarga();
        for (RAM ram : listRAM)
        {
            this->total_harga += ram.getHarga();
        }
    }

    ~Komputer() {}
};